File: go/leap/leap.go
// Package leap implements func IsLeapYear that examines if the year is a leap.
package leap

// IsLeapYear returns true if given year is a leap.
func IsLeapYear(year int) bool {
	if year%4 == 0 {
		if year%100 == 0 {
			if year%400 == 0 {
				return true
			}
			return false
		}
		return true
	}
	return false
}

File: python/leap/leap.py
"""Function for determination if the year is leap."""


def leap_year(year: int) -> bool:
    """Report if a year is leap.

    :param year: int - year.
    :return: bool
    """

    return (year % 4 == 0 and not year % 100 == 0) or year % 400 == 0
